### PR TITLE
Fix E721 warnings for .pyx files

### PR DESCRIPTION
**Pull request by jhpalmieri**
Fix E721 warnings for .pyx files

pycodestyle reports: "E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`." So replace `type(x) == str` with `isinstance(x, str)`, replace `type(x) != type(y)` with `type(x) is not type(y)`, etc. This is a followup to #36032 which dealt with .py files.

### :memo: Checklist

- [X] The title is concise, informative, and self-explanatory.
- [X] The description explains in detail what this PR is about.
- [X] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/rings/integer.pyx [hunk 1]
```diff
@@ -2852,12 +2852,12 @@ cdef class Integer(sage.structure.element.EuclideanDomainElement):
         except (ValueError, TypeError):
             pass
 
-        if type(m) == Integer and type(self) == Integer:
+        if isinstance(m, Integer) and isinstance(self, Integer):
```
Comment (mkoeppe) [inline src/sage/rings/integer.pyx]: Not the fault of this change, but the 2nd test seems redundant.

File: src/sage/rings/integer.pyx [hunk 2]
```diff
@@ -2852,12 +2852,12 @@ cdef class Integer(sage.structure.element.EuclideanDomainElement):
...
             elog = self.exact_log(m)
             if elog == -sage.rings.infinity.infinity or m**elog == self:
                 return elog
 
-        if (type(m) == Rational and type(self) == Integer
+        if (isinstance(m, Rational) and isinstance(self, Integer)
```
Comment (mkoeppe) [inline src/sage/rings/integer.pyx]: same here

File: src/sage/libs/coxeter3/coxeter.pyx [hunk 3]
```diff
@@ -86,7 +86,7 @@ cdef class String:
             sage: all([tb > ta1, tb >= ta1, tb >= tb])                 # optional - coxeter3
             True
         """
-        if type(other) != type(self):
+        if type(other) is not type(self):
```
Comment (mkoeppe) [inline src/sage/libs/coxeter3/coxeter.pyx:89]: Not the fault of this change, but this should probably do `return op == Py_NE` in this case (and possibly return `NotImplemented` for LT, LE, GT, GE)

File: src/sage/libs/coxeter3/coxeter.pyx [hunk 4]
```diff
@@ -199,7 +199,7 @@ cdef class Type:
...
```
Comment (mkoeppe) [inline src/sage/libs/coxeter3/coxeter.pyx:204]: same as above

File: src/sage/libs/coxeter3/coxeter.pyx [hunk 5]
```diff
@@ -363,7 +363,7 @@ cdef class CoxGroup(SageObject):
             sage: B4 >= A5                                                              # optional - coxeter3
             True
         """
-        if type(other) != type(self):
+        if type(other) is not type(self):
```
Comment (mkoeppe) [inline src/sage/libs/coxeter3/coxeter.pyx:370]: same as above

File: src/sage/libs/coxeter3/coxeter.pyx [hunk 6]
```diff
@@ -839,7 +839,7 @@ cdef class CoxGroupElement:
             sage: w1 != v1                                        # optional - coxeter3
             True
         """
-        if type(other) != type(self):
+        if type(other) is not type(self):
```
Comment (mkoeppe) [inline src/sage/libs/coxeter3/coxeter.pyx:848]: same as above

File: src/sage/libs/coxeter3/coxeter.pyx (hunk 3 above)
Comment (jhpalmieri) [reply src/sage/libs/coxeter3/coxeter.pyx:89]: How about
```
        if type(other) is not type(self):
            if op in (Py_LT, Py_LE, Py_GT, Py_GE):
                return NotImplemented
            return op == Py_NE
```
?

Comment (mkoeppe) [reply src/sage/libs/coxeter3/coxeter.pyx:89]: Yes, LGTM

**Comment by jhpalmieri** [comment]
Thank you for the suggestions; I've implemented them.

**Comment by jhpalmieri** [comment]
I'm not sure how much of an effect "Resolve conversation" has, but it's fun to click buttons...

**Review by mkoeppe — APPROVED** [review]

**Comment by mkoeppe** [comment]
> I'm not sure how much of an effect "Resolve conversation" has

It just collapses the comments for all readers. Good idea to do when a comment has been taken care of and you're not adding a response.

**Comment by jhpalmieri** [comment]
Thanks! I will rebase on the next beta when it's released.

**Comment by jhpalmieri** [comment]
This merges cleanly with 10.1.beta9 as is.

**Comment by vbraun** [comment]
Doesn't build, but somehow has positive review

**Comment by mkoeppe** [comment]
Indeed
```
[sagelib-10.1.beta8]     [7/9] Cythonizing sage/rings/integer.pyx
[sagelib-10.1.beta8]     [9/9] Cythonizing sage/structure/category_object.pyx
[sagelib-10.1.beta8] 
[sagelib-10.1.beta8]     Error compiling Cython file:
[sagelib-10.1.beta8]     ------------------------------------------------------------
[sagelib-10.1.beta8]     ...
[sagelib-10.1.beta8]             if isinstance(m, Integer):
[sagelib-10.1.beta8]                 elog = self.exact_log(m)
[sagelib-10.1.beta8]                 if elog == -sage.rings.infinity.infinity or m**elog == self:
[sagelib-10.1.beta8]                     return elog
[sagelib-10.1.beta8] 
[sagelib-10.1.beta8]             if (isinstance(m, Rational):
[sagelib-10.1.beta8]                                       ^
[sagelib-10.1.beta8]     ------------------------------------------------------------
[sagelib-10.1.beta8] 
[sagelib-10.1.beta8]     sage/rings/integer.pyx:2860:35: Expected ')', found ':'
[sagelib-10.1.beta8]     Traceback (most recent call last):
```

**Pull request merged**